core/cloudstorage.cpp

```diff
@@ -64,8 +65,8 @@ void CloudStorageAuthenticate::uploadFinished()
 		report_error(qPrintable(tr("Cloud account verification required, enter PIN in preferences")));
 	} else if (cloudAuthReply == QLatin1String("[PASSWDCHANGED]")) {
 		free((void *)prefs.cloud_storage_password);
-		prefs.cloud_storage_password = prefs.cloud_storage_newpassword;
-		prefs.cloud_storage_newpassword = NULL;
+		prefs.cloud_storage_password = strdup(qPrintable(cloudNewPassword));
```

Comment:
qPrefCloudStorage::set_password sounds like the better option, considering intent of making prefs readonly